lib/Core/Executor.h

```diff
@@ -164,9 +164,9 @@ class Executor : public Interpreter {
   /// globals that have no representative object (i.e. functions).
   std::map<const llvm::GlobalValue*, ref<ConstantExpr> > globalAddresses;
 
-  /// The set of legal function addresses, used to validate function
-  /// pointers. We use the actual Function* address as the function address.
-  std::set<uint64_t> legalFunctions;
+  /// Map of legal function addresses to the corresponding Function.
+  /// Used to validate and dereference function pointers.
+  std::map<std::uint64_t, llvm::Function*> legalFunctions;
```

Comment:
Since we only use `.find()` on `legalFunctions` maybe use an `unordered_map`?

Reply:
good point, I changed it